src/packages/name-service/blockstack-service.ts

```diff
@@ -143,16 +143,16 @@ export class BlockstackService extends nameService.NameService {
     }
 
     public restoreIdentity = async (fullCruxId: string, identityClaim: nameService.IIdentityClaim): Promise<nameService.IIdentityClaim> => {
-        if (!identityClaim.secrets || !identityClaim.secrets.mnemonic) {
-            throw ErrorHelper.getPackageError(PackageErrorCode.CouldNotFindMnemonicToRestoreIdentity);
-        }
-
         const mnemonic = identityClaim.secrets.mnemonic;
-        let identityKeyPair = identityClaim.secrets.identityKeyPair || undefined;
+        let identityKeyPair = identityClaim.secrets.identityKeyPair;
 
-        // If identityKeypair is not stored locally, generate them using the mnemonic
+        // If identityKeypair is not provided, generate them using the mnemonic provided
         if (!identityKeyPair) {
-            identityKeyPair = await this._generateIdentityKeyPair(mnemonic);
+            if (!mnemonic) {
+                throw ErrorHelper.getPackageError(PackageErrorCode.CouldNotFindMnemonicToRestoreIdentity);
+            } else {
+                identityKeyPair = await this._generateIdentityKeyPair(mnemonic);
```

Comment:
Is it only a safe scenario or there is a valid case when identityKeyPair might not exist as part of restoreIdentity?
I would say let restore not handle new generation and caller of it somehow make sure to call restore only if restore is possible.

Reply:
It is a safe scenario, but I would prefer to have this in restore, for future use cases of restoring using the 'mnemonic' saved by the user.

Reply:
removed the mnemonic path